service/raft/raft_group0.hh

```diff
@@ -291,6 +291,11 @@ public:
         return _raft_gr.group0_with_timeouts();
     }
 
+    // Hold shutdown gate to be waited during shutdown
+    gate::holder hold_group0_gate() {
+        return _shutdown_gate.hold();
```

Comment:
Why do you think it would? Any such case would have caused crash now (at least in debug mode). But I do not like such exception translation is a good thing. Just make it harder to understand logs.

Reply:
> Why do you think it would? Any such case would have caused crash now (at least in debug mode). 

dtest gating is not running in debug mode

> Just make it harder to understand logs.

adding more info to the exception would make it easier to understand. We can keep the old reason (gate closed) but extend it with more info (who tried to use the gate after it was closed for example)

Reply:
> > Why do you think it would? Any such case would have caused crash now (at least in debug mode).
> 
> dtest gating is not running in debug mode

Tests should not dictate how we are writing code. It should be other way around. We should definitely not trying to match our output to whatever dtest happens to be expecting. And we already have a lot of gate closed exception during shutdown, so dtest should already ignore those at shutdown time.

> 
> > Just make it harder to understand logs.
> 
> adding more info to the exception would make it easier to understand. We can keep the old reason (gate closed) but extend it with more info (who tried to use the gate after it was closed for example)

While it would be nice to add more info into the gate close exception it is out of scope here. The exception is not extensible, so we need to start from seastar.

Reply:
> Tests should not dictate how we are writing code

You yourself said that it would be a crash in debug mode, so you rely on that

Anyway, I'll just assign to you every new dtest failure with gate_closed_exception and the only thing you'll be doing for the next 2 weeks after the PR is merged is investigating them. If that's your wish

> While it would be nice to add more info into the gate close exception it is out of scope here. The exception is not extensible, so we need to start from seastar.

You can catch the exception and throw another one with more details. We're doing it in many places already

Reply:
> > Tests should not dictate how we are writing code
> 
> You yourself said that it would be a crash in debug mode, so you rely on that

What do you mean? If test crashes we need to fix it obviously.

> 
> Anyway, I'll just assign to you every new dtest failure with gate_closed_exception and the only thing you'll be doing for the next 2 weeks after the PR is merged is investigating them. If that's your wish

Fair enough :) It should not be hard to see if the failure is related to this gate.

> 
> > While it would be nice to add more info into the gate close exception it is out of scope here. The exception is not extensible, so we need to start from seastar.
> 
> You can catch the exception and throw another one with more details. We're doing it in many places already

Adding info is fine. Why change the exception type though just because the original exception cannot hold the info. If someone above in the call chain wants to catch a specific exception and you change the type it will escape for instance.

I also do not really understand why do you think that new request aborted is better than new gate closed. Both happen and reported in plenty of places during shutdown and if dtest filters are too specific they will not catch new once that are added here.